src/Agent.Listener/Configuration/NativeWindowsServiceHelper.cs

```diff
@@ -324,15 +324,14 @@ public bool IsValidCredential(string domain, string userName, string logonPasswo
 
         public NTAccount GetDefaultServiceAccount()
         {
-            SecurityIdentifier sid = new SecurityIdentifier(WellKnownSidType.LocalServiceSid, domainSid: null);
+            SecurityIdentifier sid = new SecurityIdentifier(WellKnownSidType.NetworkServiceSid, domainSid: null);
```

Comment:
LocalService/NetworkService/LocalSystem has nothing to do with domain joined or not. :)